### DISCUSSION TITLE
Is Memory.c have documented?

### DISCUSSION THREAD

**Discussion by tachibana51** [SoftEtherVPN/SoftEtherVPN#1527] [Q&A] [answered]
Hi, I'm considering fixing some memory corruption bottom up.
I noticed that the allocator of this program is very similar to old glibc malloc (probably a reimplementation of Windows).
The glibc multithreaded logic is very complex, and this allocator is also complex and confusing. There are also historical reasons specific to the program ...
I have a little knowledge about Allocator and memory safety (and some exploit in the wild). 
I'd like to read the historical background of this issue, but if we don't have an email archive or documentation at all,I think it's also necessary to gradually clarify the implementer's intent (by comments, glibc documentation, etc.).

**Comment by domosekai** [comment, marked as answer]
The author may be too busy to visit here. So any question with regard to his intention will probably not getting answered.

**Comment by tachibana51** [comment]
Thanks,
I'm writing some testcodes for this problem and writing commits.
Your recently efortance is very good and these fixes encourage me.
I'm thinking better approach for Momory.c,  If all goes well, many bugs may be fixed.
It will take a while, so please wait for a while.

**Reply by domosekai** [reply]
No problem. Take your time.
